scripts/ci/test_source.py

```diff
@@ -62,38 +61,26 @@
                f'ALL_TESTS: {ALL_TESTS}.')
 
 
-class TestExtensionSourceMeta(type):
-    def __new__(mcs, name, bases, _dict):
+class TestExtensionRecordingMode(unittest.TestCase):
 
-        def gen_test(ext_path):
-            def test(self):
-                ext_install_dir = os.path.join(self.ext_dir, 'ext')
-                pip_args = [sys.executable, '-m', 'pip', 'install', '--upgrade', '--target',
-                            ext_install_dir, ext_path]
-                check_call(pip_args)
-                unittest_args = [sys.executable, '-m', 'unittest', 'discover', '-v', ext_path]
-                env = os.environ.copy()
-                env['PYTHONPATH'] = ext_install_dir
-                env['AZURE_CORE_USE_COMMAND_INDEX'] = 'false'
-                check_call(unittest_args, env=env)
-            return test
+    def run_command(self, cmd, check_return_code=False, cwd=None):
+        logger.info(f'cmd: {cmd}')
+        out = run(cmd, check=True, cwd=cwd)
+        if check_return_code and out.returncode:
+            raise RuntimeError(f"{cmd} failed")
 
+    def test_extension(self):
```

Comment:
This method can automatically be run by `unittest.main()` because it has `test_` prefix. How about we refactor this whole `scripts/ci/test_source.py` and get rid of `unittest`?

Reply:
In the future, we need to find a way to manage all the test scripts under the scripts directory, which is too messy now.